### ISSUE TITLE
Cudnn problem with cpu backend

### ISSUE THREAD

**Issue by papperwing** [deeplearning4j/deeplearning4j#4390]
CudnnHelper used with CPU backend loaded https://gist.github.com/papperwing/60e65c45be45f84c3ee1b3d355706d86
class with exception: https://github.com/papperwing/image-net/blob/0.9.1_branch/modules/image-net-core/src/main/java/cz/muni/fi/image/net/core/objects/NeuralNetModelWrapper.java

pom.xml https://github.com/papperwing/image-net/blob/0.9.1_branch/modules/image-net-core/pom.xml

Whole project: https://github.com/papperwing/image-net/tree/0.9.1_branch

DL4J version: 0.9.1

**Comment by papperwing** [comment]
Parameters I run it with https://gist.github.com/papperwing/fbdf1dc4e27f0a35269caa9fc1f0aa99

**Comment by saudet** [comment]
That sounds pretty normal. Execution continues normally with the CPU, right?

**Comment by papperwing** [comment]
Yeah it continues but it produce like hundereds lines of warnings which is really messy.

**Comment by papperwing** [comment]
Maybe this could be marked as improvement where when there is no cuda backend used it will not even look for cudnn?
